src/foam/core/Unit.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+ foam.CLASS({
+  package: 'foam.core',
+  name: 'Unit',
+  //abstract: true,
+
+  documentation: `The abstract model for fungible digitized assets`,
+
+  ids: [
+    'alphabeticCode'
+  ],
+
+  javaImports: [
+    'foam.util.SafetyUtil'
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'name',
+      documentation: 'name of the asset',
+      required: true
+    },
+    {
+      class: 'String',
+      name: 'alphabeticCode',
+      label: 'Code',
+      documentation: 'The alphabetic code associated with the asset. Used as an ID.',
+      required: true,
+      aliases: [
+        'id'
+      ]
+    },
+    {
+      class: 'Int',
+      name: 'precision',
```

Comment:
I see an issue with this.

Since `UnitValue` extends long, all values are integers. Therefore `precision` doesn't make sense on `Unit`.

Either that or `UnitValue` should extend `Float` or `Double` instead.

Reply:
each value is stored in the system as a long, precision turns it into a decimal for view purposes. ie precision of two and UnitValue of 100 would display as 1.00

Reply:
True, forgot it was just for display. Thanks.